docs/madr/decisions/074-naming-internal-envoy-resources.md

```diff
@@ -0,0 +1,194 @@
+# Standardized Naming for internal xDS Resources
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/13266
+
+Supersedes: https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/036-internal-listeners.md
+
+## Context and Problem Statement
+
+We use resource names to identify listeners, clusters, routes, virtual hosts and secrets in the xDS APIs.
+These names appear in metrics, logs, and debug endpoints (config_dump), and they are often used for monitoring, filtering, and diagnostics.
+
+It's important to know the difference between system resources (the ones that are defined by Kuma itself, and they usually do not concern the end users) and user resources (the ones that are defined indirectly by the user and the users are interested in).
+For example, clusters like `kuma:envoy:admin` or `_kuma:dynamicconfig` are system resources, while `kri_msvc_mesh-1_us-east-2_kuma-demo_backend_httpport` cluster is user resource.
+
+Currently, there is inconsistency in how these resource names are formed, some of them have a `_kuma` prefix, some of them just contain `kuma` and some are free form.
+
+Some internal resources do not map clearly to Kuma resources in the data store, making it difficult to relate metrics to Kuma abstractions.
+
+There are also resources that seem to be in a bit of a gray area, like `meshtrace_datadog` which is a cluster created for `MeshTrace` policy.
+It could be traced down to MeshTrace resource `kri_mtr_mesh-1__kuma-system_my-meshtrace_` and if a cluster operator has problems with tracing collection it might be something to look at.
+
+To make it easier to distinguish between the two types we introduce the following definition:
+
+- **User resource** - any resource that comes from `Mesh*Service` and `Mesh*Route` definition
+- **System resource** - any resource that is not a user resource
+
+### Kuma system resource names
+
+Listeners:
+- inbound
+  - kuma:envoy:admin
+  - _kuma:dynamicconfig
+  - listener./tmp/kuma-dp-728637052/kuma-mesh-metric-config.sock
+  - listener.0.0.0.0_15001
+  - listener.0.0.0.0_15006
+  - listener.10.42.0.9_9901
+  - listener.[__]_15001 # is it ipv6?
+  - probe_listener
+  - prometheus_listener
+- outbound
+  - outbound:passthrough:ipv4
+  - outbound:passthrough:ipv6
+
+Clusters:
+- access_log_sink
+- kuma:envoy:admin
+- kuma:metrics:hijacker
+- kuma:readiness
+- meshtrace_[zipkin|datadog|otel]
+- envoy_grpc_streams (ads)
+
+Virtual Hosts:
+- _kuma:dynamicconfig
+- kuma:envoy:admin
+
+Routes (not all of them even have a name):
```

Comment:
ideally all our resources should have a name right?

Reply:
I'll add a use-case for that